File: back/dentalium/src/main/java/com/romoreno/dentalium/infraestructure/adapter/in/controller/rest/appointment/dto/SaveAppointmentRequest.java
package com.romoreno.dentalium.infraestructure.adapter.in.controller.rest.appointment.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SaveAppointmentRequest {

    private Integer id;
    private String patientId;
    private String doctorId;
    private Date since;
    private Date until;
    private String observations;
    private Integer statusCode;

//    @Data
//    @AllArgsConstructor
//    @NoArgsConstructor
//    public class Status {
//        private String code;
//        private String description;
//    }

}

File: front/dentalium/src/app/model/api/SaveAppointmentRequest.ts
export class SaveAppointmentRequest {
  patientId: string;
  doctorId: string;
  since: Date;
  until: Date;
  observations: string;
  statusCode: number;

  constructor(patientId: string, doctorId: string, since: Date, until: Date, observations: string, statusCode: number) {
    this.patientId = patientId;
    this.doctorId = doctorId;
    this.since = since;
    this.until = until;
    this.observations = observations;
    this.statusCode = statusCode;
  }
}
